WordPress/Classes/Services/AccountService.m

```diff
@@ -497,11 +495,13 @@ - (void)setVisibility:(BOOL)visible forBlogs:(NSArray *)blogs
             }
             blog.visible = visible;
         }
-    }];
-
-    [remote updateBlogsVisibility:blogVisibility success:nil failure:^(NSError *error) {
-        DDLogError(@"Error setting blog visibility: %@", error);
-    }];
+    } completion:^{
+        WPAccount *defaultAccount = [WPAccount lookupDefaultWordPressComAccountInContext:self.coreDataStack.mainContext];
```

Comment:
Would it make sense to store `defaultAccount` outside of the blocks, so that it can by the block above (line 491/489) and read her without an additional lookup?

I don't thing the account changes in the block above, but even if it did, it shouldn't be a problem since we only use it to load the `AccountServiceRemoteREST` instance.

Reply:
`NSManagedObject` instances are not thread-safe, we should not use them outside of their context queue. [This doc](https://developer.apple.com/documentation/coredata/using_core_data_in_the_background#2904037) has more (but vague IMO) info:
> Managed objects retrieved from a context are bound to the same queue that the context is bound to.

Reply:
Ah! I didn't realize `WPAccount` was an `NSManagedObject`:

https://github.com/wordpress-mobile/WordPress-iOS/blob/f8757f1ccff0a37e2aec064bc0551190aeeaa8ef/WordPress/Classes/Models/WPAccount.h#L8

Makes sense 👍 

So many gotchas using Core Data 😩 I'm so thankful for all this work your are doing pushing it more and more into a small kernel so less parts of the codebase need to know about this.